### ISSUE TITLE
Not linked with libyajl - TCL 8.5+ cannot load - Ubuntu (work-around)

### ISSUE THREAD

**Issue by tamusjroyce** [flightaware/yajl-tcl#33]
Fix: 
Including **/usr/lib/x86_64-linux-gnu/libyajl.so** (wherever this library happens to be installed...edit: dynamically discover this - thanks @resuna!) in Makefile.in
`gcc -shared -pipe -O2 -fomit-frame-pointer -Wall -fPIC  -Wl,--export-dynamic  -o libyajltcl1.6.2.so yajltcl.o tclyajltcl.o yajltcllex.o /usr/lib/x86_64-linux-gnu/libyajl.so -L/usr/lib/x86_64-linux-gnu -ltclstub8.6
`
or running the above manually before running "sudo make install"

edit:  On loading via "package require yajltcl" I would get:  yajl_gen_array_close tcl undefined error.

**Comment by resuna** [comment]
You shouldn't put platform-specific paths in Makefile.in

**Comment by tamusjroyce** [comment]
Absolutely. That is why this is a bug / work-around. To actually fix the bug, that work will need done. I may try to come back and do this (fork + pull request). But I have a product I need to develop for a customer first. 👍

**Comment by resuna** [comment]
Roger!

**Comment by gahr** [comment]
I don’t see that pkg_check_modules line being commented in configure.in

**Comment by tamusjroyce** [comment]
That was part of the work-around I needed to do. Some reason "sudo apt-get install libyajl-dev" does not add itself to pkg-config. Maybe this part of the bug really belongs in that package?

But I did read that all static libraries need linked together for TCL 8.5+. Did not used to be that way. And I did not see libyajl.so being linked in the final dynamic library.

_____________________________________________________________________________________________
tl;dr Note: Using Bash on Windows -> Ubuntu with Netbeans + http://plugins.netbeans.org/plugin/39119/tcl-plugin + vxcsrv. But works same as Ubuntu w/ KDE. Wasn't able to get that far with Msys2 / MinGW. But the system I need to develop for is embedded Linux based using https://github.com/flightaware/yajl-tcl for a certain remote control TCL 2.0 software.

This library is really useful! Thank you to those who wrote it and those supporting it. Including @gahr  and @resuna!

**Comment by gahr** [comment]
I’m not an ubuntu expert, but the pkgconf file is listed here: https://packages.ubuntu.com/bionic/amd64/libyajl-dev/filelist

**Comment by tamusjroyce** [comment]
I should technically be on Ubuntu 16.04. But it is Bash on Windows. So anything could be causing the pkg-config issue. Good to know! I will remove that part / step #1 from this bug.

@mutability - I see that you are correct. I should have looked...my apologies!
prefix=/usr
libdir=${prefix}/lib/x86_64-linux-gnu
includedir=${prefix}/include/yajl

Name: Yet Another JSON Library
Description: A Portable JSON parsing and serialization library in ANSI C
Version: **2.1.0**
Cflags: -I${includedir}
Libs: -L${libdir} -lyajl
